Lib/fontTools/ttLib/tables/_g_v_a_r.py

```diff
@@ -39,6 +39,10 @@
 GVAR_HEADER_SIZE = sstruct.calcsize(GVAR_HEADER_FORMAT)
 
 class _LazyDict(UserDict):
+	def __init__(self, data):
+		super().__init__()
+		self.data = data
```

Comment:
I don't think you need to override `__init__`, just pass the dict to the UserDict's default constructor and it will populate its internal self.data attribute with it:

https://github.com/python/cpython/blob/b69ba3d7233013ea8cb95a3e30e2338f7f76b359/Lib/collections/__init__.py#L1088-L1096

Reply:
oh, maybe the `update` method that is called by the default constructor un-lazifies the data which defeats the purpose of the lazy dict?

Reply:
I just tried to delete the custom `_LazyDict.__init__`, and checked if the inner update() would evaluate the values, but it seems that it doesn't:

```python
In [1]: from fontTools.ttLib.tables._g_v_a_r import _LazyDict
In [2]: d = _LazyDict({"a": lambda: print("HEY!!!") or 1})  # initializing the lazy dict prints nothing
In [3]: d["a"]
HEY!!!
Out [3]: 1
```

Reply:
A custom __init__ indeed isn't needed, but I thought it's a bit wasteful to go through UserDict's __setitem__ if we know it just has to end up in data anyway.

Reply:
hm ok. Maybe write a comment along those lines.
Usually doing `dict(d)` creates a shallow copy of the otherdict (which is what the default UserDict constructor also does), whereas in this case _LazyDict just sets its internal data to refer to that other dict. I suppose it's fine since this module is only consumer, but if this were a public LazyDict class making the copy would be more in line with the dict-like objects behavior.

Reply:
Look how I wrote it before: https://github.com/fonttools/fonttools/pull/2781/commits/cd68be6188b4c2711b1f0e47f7e4881d666a8356
Perhaps that makes it more explicit that it's a bit of a hack to avoid some unnecessary overhead.

Reply:
you mean reverting to the first commit where you set data attribute after the fact? hm yeah if you like.
Or an alternative classmethod constructor like `self.variations = _LazyDict.with_data(variations)` so it's on one line.
In the end, it doesn't really matter because this it's only used once here.